File: 02_Nested_Loops.cpp
#include <iostream>
#include <vector>
using namespace std;

void simulateNestedLoops(int n, vector<int>& currentLoop) {
    if (currentLoop.size() == n) {
        for (int i = 0; i < n; i++) {
            cout << currentLoop[i] << " ";
        }
        cout << endl;
        return;
    }

    for (int i = 1; i <= n; i++) {
        currentLoop.push_back(i);
        simulateNestedLoops(n, currentLoop);
        currentLoop.pop_back();
    }
}

int main() {
    int n;
    cout << "Въведете стойността на n: ";
    cin >> n;

    vector<int> currentLoop;
    simulateNestedLoops(n, currentLoop);

    return 0;
}

File: 02_Nested_Loops.py
def simulate_nested_loops(n, current_loop=[]):
    if len(current_loop) == n:
        print(*current_loop)
        return

    for i in range(1, n + 1):
        simulate_nested_loops(n, current_loop + [i])


n = int(input("Въведете стойността на n: "))
simulate_nested_loops(n)
